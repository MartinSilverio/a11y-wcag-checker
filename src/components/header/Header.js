import React from 'react';
import MenuIcon from '@material-ui/icons/Menu';
import AccessibilityNewIcon from '@material-ui/icons/AccessibilityNew';
import { useDispatch } from 'react-redux';
import { toggleNavMenu } from '../../redux/nav/navActions';

import './Header.scss';

function Header() {
  const dispatch = useDispatch();

  function handleDrawerToggle() {
    dispatch(toggleNavMenu());
  }

  return (
    <header id='root-header'>
      {/* <Toolbar>
        <Hidden smUp>
          <IconButton
            color='inherit'
            aria-label='open drawer'
            edge='start'
            onClick={handleDrawerToggle}
            className={classes.menuButton}
          >
            <MenuIcon />
          </IconButton>
        </Hidden>
        <AccessibilityNewIcon aria-hidden={true} fontSize='large' />
        <Typography variant='h1'>A11y-Checker</Typography>
      </Toolbar> */}
      <button className='mobile-open-menu' onClick={handleDrawerToggle}>
        <i className='fas fa-bars'></i>
      </button>
      <h1>A11y-Checker</h1>
    </header>
  );
}

export default Header;
