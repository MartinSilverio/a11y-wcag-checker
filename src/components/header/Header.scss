@import '../../colors';

#root-header {
  display: flex;
  // border: 2px black dashed;
  background-color: $primary;
  padding-left: 20px;
  // box-shadow: 0 10px #6161b1;

  box-shadow: 0 5px 5px rgba($color: $primary, $alpha: 0.4);
  z-index: 2;

  & > .mobile-open-menu {
    display: none;
  }

  /* Portrait and Landscape */
  @media only screen and (max-width: 812px) {
    padding-left: 0;

    & > .mobile-open-menu {
      display: block;
      background-color: transparent;
      border: none;
      font-size: 24px;
      width: 60px;
    }
  }
}
