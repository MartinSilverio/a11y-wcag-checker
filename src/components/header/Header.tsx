import React from 'react';
// @ts-expect-error ts-migrate(7016) FIXME: Could not find a declaration file for module 'reac... Remove this comment to see the full error message
import { useDispatch } from 'react-redux';
import { toggleNavMenu } from '../../redux/nav/navActions';
import './Header.scss';

function Header() {
  const dispatch = useDispatch();

  function handleDrawerToggle() {
    dispatch(toggleNavMenu());
  }

  return (
    // @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message
    <header id='root-header'>
      {/* @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message */}
      <button className='mobile-open-menu' onClick={handleDrawerToggle}>
        {/* @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message */}
        <i className='fas fa-bars'></i>
      </button>
      {/* @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message */}
      <h1>A11y-Checker</h1>
    </header>
  );
}

export default Header;
