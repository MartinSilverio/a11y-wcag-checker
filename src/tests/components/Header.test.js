import React from 'react';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import Header from '../../components/header/Header';
import NavActionTypes from '../../redux/nav/navTypes';

const mockStore = configureMockStore();

describe('Header', () => {
  let headerWrapper;
  let store = mockStore();
  beforeEach(() => {
    headerWrapper = mount(
      <Provider store={store}>
        <Header />
      </Provider>
    );
  });

  afterEach(() => {
    headerWrapper.unmount();
  });

  it('should match the snapshot', () => {
    expect.assertions(1);
    expect(headerWrapper.find('Header').html()).toMatchSnapshot();
  });

  it('should fire action with correct type', () => {
    expect.assertions(1);
    const expectedAction = {
      type: NavActionTypes.TOGGLE_NAV_MENU,
    };

    headerWrapper.find('button').simulate('click');
    let actions = store.getActions();
    expect(actions[0]).toEqual(expectedAction);
  });
});
