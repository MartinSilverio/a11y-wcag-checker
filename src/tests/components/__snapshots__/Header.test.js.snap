// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match the snapshot 1`] = `"<header id=\\"root-header\\"><button class=\\"mobile-open-menu\\"><i class=\\"fas fa-bars\\"></i></button><h1>A11y-Checker</h1></header>"`;
